### PR TITLE
Version: 1.7.0 [Architecture Rework Implementations]

### PR DESCRIPTION
**Pull request by victoriaquasar**
- ☢ Breaking change
- 📌 New feature/implementation 

### 🚀 **Tests Development**

We are developing unit tests to cover misbehavior and bugs within Melon so we will be able to identify, prevent and fix internal issues with our libraries and core methods, we have also been able to resolve several minor issues identified and we are excited to end full test coverage. In the next minor versions we will be focused in bug bounty and aim the full coverage of Melon.

### ⚡ **Hard Optimizations**

Several internal optimizations were performed, such as the implementation of internal lazy loading, reduction of steps to achieve the construction of the main Melon Engine instance and complete reorganization of the solution structure, with a focus on architectural scalability.

- **[core]** Csharp solution rebuilt focusing in best pratices and scalability
- **[core]** Implemented lazy loading for internal JS libraries + builder pattern for creating the final Engine instance
- **[core]** Fixed optimization gaps in the code (JavaScript and C#)

### ♻ **Rework Implementations**

- 📌 **[core] [new]** Created new `guards` library to make easier declarative programming with Melon
- 📌 **[core] [new]** Implemented .NET interop with `xrequire("dotnet:[MODULE]")`
- 📌 **[core] [new]** Implemented .NET external assembly loading with `xrequire("dotnet-external:[PATH]")`
- ☢ **[core]** Re-implemented `std` library from zero with several changes
- 📌 **[core]** Implemented wrapped `Promise` as content of `std` to ease imports
- 📌 **[core]** Re-implemented `std.shift()` with the same behavior as the old one
- 📌 **[core]** Implemented `system` as sub-module with its own methods
- 📌 **[core]** Implemented `environment` as sub-module with its own methods
- 📌 **[core]** Implemented `process` as sub-module with its own methods
- ☢ **[core]** Re-implemented `http` library from zero with several changes
- 📌 **[core]** Implemented wrapped `HttpRoute` as content of `std` to ease imports
- 📌 **[core]** Implemented wrapped `HttpApplication` as content of `std` to ease imports
- 📌 **[core]** Implemented wrapped `Response` as content of `std` to ease imports
- 📌 **[core]** Implemented `http.[result/static/image/audio/video/pdf/html]` as valid webapp route responses
- ☢ **[core]** Re-implemented commands with only `--silent` and `--disable[target]`

### CODE REVIEW

File: projects/melon-runtime/Melon.Engine/Builder/EngineBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            var content = LibraryLoader.ByIdentifier(identifier);
+            _engine.Execute(content);
+
+            return this;
+        }
+
+        public Jint.Engine Build()
+        {
+            return _engine;
+        }
```
Comment (vilanz) [inline projects/melon-runtime/Melon.Engine/Builder/EngineBuilder.cs:28]: Minor detail: `.Build()` just returning the internal Jint instance is a bit weird.
I'd expect to have a method like `.AddLibrary(item)` to just append the libraries to the list, and `.Build()` just loads each item in the list.

File: projects/melon-runtime/Melon.Library/Data/data.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
+﻿const data = {
+    clone: function (target) {
+        if (typeof target === "function") {
+            return function () { target(...arguments ?? ''); }
+        }
+
+        return JSON.parse(JSON.stringify(eval(target)));
```
Comment (vilanz) [inline projects/melon-runtime/Melon.Library/Data/data.js:7]: I see the `eval` there :) Why is it necessary? Isn't target possibly a value that can be stringified?

File: projects/melon-types/AsyncLoop.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+declare class AsyncLoop<T> {
+    constructor(action: Function, times?: number, delay?: number, errorAction?: Function);
+    stop: () => void;
+    times: number;
+    execute: () => Promise<T>;
+}
```
Comment (vilanz) [inline projects/melon-types/AsyncLoop.d.ts:6]: Any news on the effort of the TypeScript build being directly inside `melon-runtime`? I don't see any issues with it being a pre-build step which generates `.js` before we do anything.
If it's truly not something viable right now, an improvement for better TypeScript integration would be to have these `d.ts` files inside `melon-runtime` somehow.
I know we're tied to .NET's JS shenanigans but just having them as "extra files" living together with JS that are copied over would already be nice. Only thing that matters is them being a folder that package.json's typings direct to.

File: projects/melon-runtime/Melon.Engine/Builder/EngineBuilder.cs (hunk 1 above)
Comment (victoriaquasar) [reply projects/melon-runtime/Melon.Engine/Builder/EngineBuilder.cs]: 3d4a917a32ab9bb5d3e9af303c9101d379a520c6

File: projects/melon-types/AsyncLoop.d.ts (hunk 3 above)
Comment (victoriaquasar) [reply projects/melon-types/AsyncLoop.d.ts:6]: Not really

File: projects/melon-runtime/Melon.Library/Data/data.js (hunk 2 above)
Comment (victoriaquasar) [reply projects/melon-runtime/Melon.Library/Data/data.js]: dfd1c923e08dcc90ab2b85a8c22de3be030db79d

**Pull request merged**